src/ipc/ipc4/handler.c

```diff
@@ -315,9 +313,6 @@ int ipc4_pipeline_prepare(struct ipc_comp_dev *ppl_icd, uint32_t cmd)
 		case COMP_STATE_INIT:
 			tr_dbg(&ipc_tr, "pipeline %d: reset from init", ppl_icd->id);
 			ret = ipc4_pipeline_complete(ipc, ppl_icd->id, cmd);
-			if (ret < 0)
-				ret = IPC4_INVALID_REQUEST;
```

Comment:
I think a `return IPC4_x` here and in all other places that are modified in this patch has 2 important advantages: (1) you can return different error codes and print different error messages, (2) when reading you see immediately what happens in error case, instead of having to check 30 lines down